src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java

```diff
@@ -0,0 +1,148 @@
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement();
+             ResultSet rs = stmt.executeQuery("SELECT * FROM FLIGHTS  ")) {
+            while (rs.next()) {
+                Long id = rs.getLong("id");
+                Long price = rs.getLong("initialPrice");
+                LocalDateTime departureDate = rs.getTimestamp("departureDate").toLocalDateTime();
+                LocalDateTime arrivalDate = rs.getTimestamp("arrivalDate").toLocalDateTime();
+                Long plane_id = rs.getLong("plane_id");
+                Plane plane = planeDAO.getById(plane_id);
+                Boolean isDeleted = rs.getBoolean("isDeleted");
+                if (!isDeleted)
+                    flights.add(Flight.builder().id(id).initialPrice(price).plane(plane).
+                            departureDate(departureDate).arrivalDate(arrivalDate).build());
+
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return flights;
+    }
+
+    @Override
+    public Flight getFlightById(Long id) {
+        Flight flight = new Flight();
+        String sql = String.format("SELECT * FROM FLIGHTS WHERE id = '%d' AND isDeleted=FALSE", id);
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement();
+             ResultSet rs = stmt.executeQuery(sql)) {
+            if (rs.next()) {
+                Long price = rs.getLong("initialPrice");
+                LocalDateTime departureDate = rs.getTimestamp("departureDate").toLocalDateTime();
+                LocalDateTime arrivalDate = rs.getTimestamp("arrivalDate").toLocalDateTime();
+                Long plane_id = rs.getLong("plane_id");
+                Plane plane = planeDAO.getById(plane_id);
+                return Flight.builder().id(id).plane(plane).initialPrice(price).
+                        departureDate(departureDate).arrivalDate(arrivalDate).build();
+
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return flight;
+    }
+
+    @Override
+    public Flight addNewFlight(Flight flight) {
+        Long price = flight.getInitialPrice();
+        LocalDateTime depatureDate = flight.getDepartureDate();
+        LocalDateTime arrivalDate = flight.getArrivalDate();
+        Long plane_id = flight.getPlane().getId();
+
+        String sql = String.format("INSERT INTO Flight (initialPrice, plane_id, departureDate,arrivalDate,isDeleted)" +
+                " VALUES (%d,%d,'%s','%s','%s')", price, plane_id, DateFormatter.format(depatureDate), DateFormatter.format(arrivalDate), "FALSE");
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                ResultSet rs = stmt.executeQuery("SELECT * FROM FLIGHTS");
```

Comment:
you can get id from executing method. Nastya did it yesterday

Reply:
If Nastya did it, i have updated it today, by merging her branch. So it is done